### ISSUE TITLE
3D Bounding box

### ISSUE THREAD

**Issue by Div99** [Microsoft/AirSim#1791]
Hi, can the PythonClient be extended to have a getBoundingBox3D method for other cars, pedestrians in the environment? 

This functionality will go a long way in making AirSim a more complete simulator, allowing for functionality such as perception of objects and prediction of their trajectories, which are standard components of most AV system pipelines.

**Comment by madratman** [comment]
Yes, technically it can be, but we don't have the bandwidth to add this as of now. 
One thing to look at is how carla does it here https://github.com/carla-simulator/carla/search?q=bounding+box&type=Code and see if you can port it to airsim.

**Comment by Div99** [comment]
Hi, I want to use the feature with pre-built environments like CityEnviron or AirSimNH. From what I understand, any changes I make won't work with the supplied binaries of environments.

**Comment by ducmanhnguyen** [comment]
Have anyone success to get 3d bounding box with lidar data for deep learning? It would be great if you can share the data or the code to generate it. Thanks :)

**Comment by ironclownfish** [comment]
In the meantime here's a ridculous hack:
You can measure the radius of an object by running it into a wall and noting the object's position relative to the position of the wall. Do this in all 6 directions and you have a bounding box. (This process could be automated, client side)

It would be much better for us to add the feature of course.
I just always try to suggest it's possible to do things client side, even if in a very weird way.

**Comment by saihv** [comment]
@ironclownfish Is there a way to extract the dimensions of the 'simple' collision box, perhaps?

**Comment by chongjeelee** [comment]
hi,guys,does solve this problem for geting 3d bounding box?
if not in airsim api,how can i do something with UE4 bluerprint ?

**Comment by chongjeelee** [comment]
i found that we can set lidar and cam on the same position(settings.json),and the cam can get 3d bounding box from environment!so this 3d bounding box can be used to lidar !
maybe can close this issue...
